src/pathfinder.rs

```diff
@@ -0,0 +1,367 @@
+        return Err(RegisteringNodesError::NodesNotCorrectlyIndexed);
+    }
+
+    // Make sure every connection we have we other nodes is 2 ways
+    new_node.connected_nodes_id.iter().for_each(|index| {
+        if let Some(node_container) = nodes.get_mut(*index) {
+            if let Some(node) = node_container.node.write().unwrap().as_mut() {
+                node.connected_nodes_id.push(new_node.unique_id)
+            }
+        };
+    });
+
+    nodes.push(NodeContainer::new(new_node));
+
+    Ok("1".to_string())
+}
+
+enum DeleteNodeError {
+    ParsingError(ParseIntError),
+    MutexError(mut_static::Error),
+    NodeNotFound,
+}
+
+impl From<ParseIntError> for DeleteNodeError {
+    fn from(error: ParseIntError) -> Self {
+        DeleteNodeError::ParsingError(error)
+    }
+}
+
+impl From<mut_static::Error> for DeleteNodeError {
+    fn from(error: mut_static::Error) -> Self {
+        DeleteNodeError::MutexError(error)
+    }
+}
+
+impl std::fmt::Display for DeleteNodeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DeleteNodeError::MutexError(e) => write!(f, "Mutex error : {e}"),
+            DeleteNodeError::ParsingError(e) => write!(f, "Parsing error : {e}"),
+            DeleteNodeError::NodeNotFound => write!(f, "Node not found"),
+        }
+    }
+}
+
+byond_fn!(fn remove_node_astar(unique_id) {
+    let mut nodes = NODES.write().unwrap();
+    match remove_node(unique_id, nodes.as_mut()) {
+        Ok(s) => Some(s),
+        Err(e) => Some(format!("{e}"))
+    }
+});
+
+// Replace the node with unique_id by None
+// Update connected nodes as well so nothing target the removed node anymore
+// Errors if no node can be found with unique_id
+fn remove_node(unique_id: &str, nodes: &mut [NodeContainer]) -> Result<String, DeleteNodeError> {
+    let unique_id = unique_id.parse::<usize>()?;
+
+    {
+        let node_to_delete_container = nodes.get(unique_id).cloned();
+
+        let node_to_delete_ref = match node_to_delete_container {
+            None => return Err(DeleteNodeError::NodeNotFound),
+            Some(node_container) => node_container.node,
+        };
+
+        let node_to_delete_ref = node_to_delete_ref.as_ref().read().unwrap();
+
+        let node_to_delete = match node_to_delete_ref.as_ref() {
+            None => return Err(DeleteNodeError::NodeNotFound),
+            Some(node) => node,
+        };
+
+        // Erase all links to the removed node
+        node_to_delete.connected_nodes_id.iter().for_each(|i| {
+            if let Some(node_container) = nodes.get_mut(*i) {
+                if let Some(node) = node_container.node.write().unwrap().as_mut() {
+                    node.connected_nodes_id
+                        .retain(|index| index != &node_to_delete.unique_id);
+                }
+            }
+        });
+    } // We need to drop everything before set the removed node to None. This is to ensure memory safety
```

Comment:
Woah woah woah, *memory safety*? None of this code is unsafe, what are we worrying about?